frontend/composables/useSimpleAsyncData.ts

```diff
@@ -0,0 +1,15 @@
+interface SimpleAsyncDataReturn<T> {
+  data: Ref<T>
+  refresh: () => Promise<void>
+}
+
+export const useSimpleAsyncData = async <T>(key: string, fn: () => Promise<T>): Promise<SimpleAsyncDataReturn<T>> => {
+  const { loading: { withLoading } } = useModal()
+  const fnWithLoading = () => withLoading(fn, key)
+  const { data: dataRef, refresh, error } = await useAsyncData(key, fnWithLoading)
+  if (error.value) {
+    throw createError(error.value)
+  }
+  const data = dataRef as unknown as Ref<T>
```

Comment:
Lol I remember saying "the trickiest part is getting the typings right", which I see borne out here.

More seriously though, are we confident that we're not losing something in translation here? I can see the "eliding via `unknown`" as causing difficult to debug typing issues.

Part of that is mitigated because you don't implement the full `useAsyncData` interface, which has things like `transform` and `pick` that complicate the types (and indeed, the actual return values) quite a bit.

Reply:
Good call - the unknown isn't nescessary, so I removed it.